uaclient/update_info.py

```diff
@@ -0,0 +1,90 @@
+from typing import Any, Dict
+
+from uaclient import util
+from uaclient.config import UAConfig
+
+# We are doing this because we are sure that python3-apt comes with the distro,
+# but it cannot be installed in a virtual environment to be properly tested.
+# All our unit tests are mocking the Cache object, so we are not losing
+# anything here. If by any chance there is no python3-apt in the future, the
+# integration tests should be able to detect it through failure, because we
+# don't expect an empty apt cache there.
+try:
+    from apt import Cache  # type: ignore
+except ImportError:
+
+    def Cache():
+        return []
```

Comment:
Ack that the suggestion below is basically the same thing, but I think I prefer it because (1) it keeps the test mock in a test file, and (2) it won't require that we add more mock code to this file if we start to rely on additional members of `apt` in the future.

```diff
diff --git a/uaclient/conftest.py b/uaclient/conftest.py
index 5328878d..1fb3fa7d 100644
--- a/uaclient/conftest.py
+++ b/uaclient/conftest.py
@@ -1,5 +1,6 @@
 import io
 import logging
+import sys
 from typing import Any, Dict
 
 import mock
@@ -7,6 +8,10 @@ import pytest
 
 from uaclient.config import UAConfig
 
+# We are doing this because we are sure that python3-apt comes with the distro,
+# but it cannot be installed in a virtual environment to be properly tested.
+sys.modules["apt"] = mock.MagicMock()
+
 
 @pytest.fixture
 def caplog_text(request):
diff --git a/uaclient/update_info.py b/uaclient/update_info.py
index 3d9b9c88..2a981d2a 100644
--- a/uaclient/update_info.py
+++ b/uaclient/update_info.py
@@ -3,18 +3,7 @@ from typing import Any, Dict
 from uaclient import util
 from uaclient.config import UAConfig
 
-# We are doing this because we are sure that python3-apt comes with the distro,
-# but it cannot be installed in a virtual environment to be properly tested.
-# All our unit tests are mocking the Cache object, so we are not losing
-# anything here. If by any chance there is no python3-apt in the future, the
-# integration tests should be able to detect it through failure, because we
-# don't expect an empty apt cache there.
-try:
-    from apt import Cache  # type: ignore
-except ImportError:
-
-    def Cache():
-        return []
+from apt import Cache  # type: ignore
 
 
 ESM_SERVICES = ("esm-infra", "esm-apps")
```

Reply:
Oh this true - I was able to reproduce it working. Thanks for reminding me of this. I will update the code to use this approach.